conn.go

```diff
@@ -324,8 +323,7 @@ func (c *Conn) closeWithError(err error) {
 	if err != nil {
 		// we should attempt to deliver the error back to the caller if it
 		// exists
-		for id := 0; id < len(c.calls); id++ {
-			req := &c.calls[id]
+		for _, req := range c.calls {
```

Comment:
I might have clicked cancel by accident making this comment earlier.  Because the slice is holding pointers now, and most of them will be nil, we will need to check if the req is nil before attempting to send on the channel.